ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -194,6 +199,8 @@ public class DagUtils {
    */
   private final ConcurrentHashMap<String, Object> copyNotifiers = new ConcurrentHashMap<>();
 
+  private static final Cache<String, String> shaCache = CacheBuilder.newBuilder().maximumSize(100).build();
```

Comment:
i don't think it's a good idea to have cache as part of util class

Reply:
this is tricky: we use this class as an instance (DagUtils.getInstance) and some classes in the hive code own an instance of this class, not as a static utility, so cache is not a problem from this point of view
having it as a static might be confusing, but it wants to be shared across different sessions as SHA computation is expensive

Reply:
what I am saying is this cache should be tight to some TezSession class like TezSessionState and not the util class

Reply:
```
what I am saying is this cache should be tight to some TezSession class like 
```
not necessarily: this cache connects local file's path to a SHA hash, which is independent of a tez session...if we connect it to a TezSessionState object, then we'll lose the already costly calculates hashes...this cache was a static one even before this patch, so it's behavior is not changed
would it make sense to make a code comment on this?

Reply:
logically connect, localization happens in TezSessionState, I think the original place was more appropriate than the util class